src/component/MovieList.js

```diff
@@ -21,16 +24,32 @@ const MovieList = () => {
         dispatch(getMovies(page))
     }
 
+    const handleChange = event => {
+        const { value: nextValue } = event.target;
+		setValue(nextValue);
+		const debouncedSave = debounce(() => {
```

Comment:
You can create separate function for these dispatches.

```
const searchMovies = () => {
    if(nextValue === '') {
        dispatch(getMovies(1))
    } else {
        dispatch(getMoviesByTitle(1, nextValue))
    }
}
```

and then call it inside debounce:

`const debouncedSave = debounce(searchMovies)`